src/theory/partition_generator.cpp

```diff
@@ -0,0 +1,169 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Amalee Wilson, Andrew Wu
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2021 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * PartitionGenerator for creating partitions.
+ */
+
+#include "theory/partition_generator.h"
+
+#include <math.h>
+
+#include "expr/node_algorithm.h"
+#include "expr/node_builder.h"
+#include "options/parallel_options.h"
+#include "prop/prop_engine.h"
+#include "prop/theory_proxy.h"
+#include "prop/zero_level_learner.h"
+#include "theory/theory_engine.h"
+#include "theory/theory_id.h"
+#include "theory/theory_traits.h"
+
+using namespace std;
+
+namespace cvc5::internal {
+
+namespace theory {
+PartitionGenerator::PartitionGenerator(Env& env,
+                                       TheoryEngine* theoryEngine,
+                                       prop::PropEngine* propEngine)
+    : EnvObj(env),
+      d_numPartitions(options().parallel.computePartitions),
+      d_numChecks(0),
+      d_numPartitionsSoFar(0)
+{
+  d_valuation = std::make_unique<Valuation>(theoryEngine);
+  d_propEngine = propEngine;
+
+  d_conflictSize = options().parallel.partitionConflictSize;
+  if (!d_conflictSize)
+  {
+    d_conflictSize = static_cast<uint64_t>(log2(d_numPartitions));
+  }
+}
+
+std::vector<TNode> PartitionGenerator::collectDecisionLiterals()
+{
+  std::vector<TNode> literals;
+  std::vector<Node> decisionNodes = d_propEngine->getPropDecisions();
+  // Make sure the literal does not have a boolean term or skolem in it.
+  const std::unordered_set<Kind, kind::KindHashFunction> kinds = {
+      kind::SKOLEM, kind::BOOLEAN_TERM_VARIABLE, kind::CONST_BOOLEAN};
```

Comment:
I don't think CONST_BOOLEAN needs to be in this set. Why is this used?

Reply:
As far as I know, `true` and `false` actually should never appear in asserted literals.  Do you know if there was a case where this was necessary?

Reply:
Different partitioning strategies use variables from different places: some from the decision trail in the sat solver, some from the order_heap (I don't think that there are other places right now). 

There was a case where this change was necessary, but I forget the details.